pkg/networkserver/downlink.go

```diff
@@ -1288,7 +1312,7 @@ func (ns *NetworkServer) processDownlinkTask(ctx context.Context) error {
 
 							if len(genState.ApplicationDownlink.GetClassBC().GetGateways()) > 0 &&
 								allErrors(nonRetryableFixedPathGatewayError, pathErrs...) {
-								logger.WithField("retry_at", nextDownlinkAt).Warn("Fixed paths invalid, fail downlink and retry downlink slot")
+								logger.WithField("retry_at", nextDownlinkAt).Warn("Fixed paths invalid, fail application downlink and retry attempt")
```

Comment:
Why still retry here? The errors are all non-retryable right?

Reply:
Also this block is not tested

Reply:
The downlink causing the non-retryable errors is dropped(and AS is informed about it), but NS still tries to schedule a (different) downlink at a later time, if there is a downlink to schedule.